### PR TITLE
Blazor Preview 9 updates

### LINKED ISSUE #14146: OnAfterRenderAsync(bool firstRender)
`OnAfterRender()` and  `OnAfterRenderAsync()` are now `OnAfterRender(bool firstRender)` and `OnAfterRenderAsync(bool firstRender)`.

### LINKED ISSUE #14155: Blazor Preview 9 updates
Mop-up work following Preview 9 release.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #14155
Fixes #14146

* Mop-up PR for Pre9.
* Recommend quick review+merge ... *There be 🐉 in our current content.*
* :eyes: *Watch me!* :eyes: on the new `firstRender` setups here. New to me and 🤔 guessing a bit.
* Updating the `ms.date` values to today.

### CODE REVIEW

File: aspnetcore/blazor/components.md [hunk 1]
```diff
@@ -708,19 +708,28 @@ protected override void OnParametersSet()
 }
 ```
 
-`OnAfterRenderAsync` and `OnAfterRender` are called after a component has finished rendering. Element and component references are populated at this point. Use this stage to perform additional initialization steps using the rendered content, such as activating third-party JavaScript libraries that operate on the rendered DOM elements.
+`OnAfterRenderAsync` and `OnAfterRender` are called after a component has finished rendering. Element and component references are populated at this point. Use this stage to perform additional initialization steps using the rendered content, such as activating third-party JavaScript libraries that operate on the rendered DOM elements. The `firstRender` parameter is:
```
Comment (danroth27) [inline aspnetcore/blazor/components.md]: We need to state here that `OnAfterRender` is *not* called when prerendering on the server.

File: aspnetcore/blazor/dependency-injection.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Services can be configured with the lifetimes shown in the following table.
 
 | Lifetime | Description |
 | -------- | ----------- |
-| <xref:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.Scoped*> | Blazor client-side doesn't currently have a concept of DI scopes. `Scoped`-registered services behave like `Singleton` services. However, the server-side hosting model supports the `Scoped` lifetime. In a Razor component, a scoped service registration is scoped to the connection. For this reason, using scoped services is preferred for services that should be scoped to the current user, even if the current intent is to run client-side in the browser. |
+| <xref:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.Scoped*> | Blazor client-side doesn't currently have a concept of DI scopes. `Scoped`-registered services behave like `Singleton` services. However, the server-side hosting model supports the `Scoped` lifetime. In a Razor component, a scoped service registration is scoped to the *connection*. For this reason, using scoped services is preferred for services that should be scoped to the current user, even if the current intent is to run client-side in the browser. |
```
Comment (danroth27) [inline aspnetcore/blazor/dependency-injection.md]:
```suggestion
| <xref:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.Scoped*> | Blazor WebAssembly apps don't currently have a concept of DI scopes. `Scoped`-registered services behave like `Singleton` services. However, the server-side hosting model supports the `Scoped` lifetime. In Blazor Server apps, a scoped service registration is scoped to the *connection*. For this reason, using scoped services is preferred for services that should be scoped to the current user, even if the current intent is to run client-side in the browser. |
```

File: aspnetcore/blazor/layouts.md [hunk 3]
```diff
@@ -61,7 +73,7 @@ The *MasterListLayout.razor* file provides the `MasterListLayout`. The layout re
 
 [!code-cshtml[](layouts/sample_snapshot/3.x/MasterListLayout.razor?highlight=1,9)]
 
-Finally, `MasterLayout` in *MasterLayout.razor* contains the top-level layout elements, such as the header, main menu, and footer. `MasterListLayout` with `EpisodesComponent` are rendered where `@Body` appears:
+Finally, `MasterLayout` in *MasterLayout.razor* contains the top-level layout elements, such as the header, main menu, and footer. `MasterListLayout` with `EpisodesComponent` is rendered where `@Body` appears:
```
Comment (danroth27) [inline aspnetcore/blazor/layouts.md]:
```suggestion
Finally, `MasterLayout` in *MasterLayout.razor* contains the top-level layout elements, such as the header, main menu, and footer. `MasterListLayout` with the `EpisodesComponent` is rendered where `@Body` appears:
```

File: aspnetcore/blazor/routing.md [hunk 4]
```diff
@@ -24,8 +24,6 @@ Blazor server-side is integrated into [ASP.NET Core Endpoint Routing](xref:funda
 
 The `Router` component enables routing, and a route template is provided to each accessible component. The `Router` component appears in the *App.razor* file:
```
Comment (danroth27) [inline aspnetcore/blazor/routing.md]:
```suggestion
The `Router` component enables routing to each component with a specified route. The `Router` component appears in the *App.razor* file:
```

File: aspnetcore/blazor/routing.md [hunk 5]
```diff
@@ -37,7 +35,14 @@ In a Blazor server-side or client-side app:
 </Router>
 ```
 
-When a *.razor* file with an `@page` directive is compiled, the generated class is provided a <xref:Microsoft.AspNetCore.Mvc.RouteAttribute> specifying the route template. At runtime, the router looks for component classes with a `RouteAttribute` and renders the component with a route template that matches the requested URL.
+When a *.razor* file with an `@page` directive is compiled, the generated class is provided a <xref:Microsoft.AspNetCore.Mvc.RouteAttribute> specifying the route template.
+
+At runtime, the `RouteView` component:
+
+* Receives the `RouteData` from the `Router` along with any desired parameters.
+* Searches for component classes with a `RouteAttribute` and renders the component with a route template that matches the requested URL.
```
Comment (danroth27) [inline aspnetcore/blazor/routing.md]: The `RouteView` component doesn't search for components. The `RouteData` passed to the `RouteView` from the `Router` already specifies a specific component and the matched parameters. All the `RouteView` does is render the specified component with its layout (or an optional default layout) using the specified parameters.

File: aspnetcore/includes/blazor-prerendering.md [hunk 6]
```diff
@@ -46,7 +49,7 @@ Where `JSRuntime.InvokeAsync` is called, `ElementRef` is only used in `OnAfterRe
     private string infoFromJs;
     private ElementReference myElem;
 
-    protected override async Task OnAfterRenderAsync()
+    protected override async Task OnAfterRenderAsync(bool firstRender)
     {
         // TEMPORARY: Currently we need this guard to avoid making the interop
```
Comment (danroth27) [inline aspnetcore/includes/blazor-prerendering.md]: This temporary code should be removed. `OnAfterRenderAsync` is no longer called during prerendering and the `IComponentContext` service has been removed.

**Comment by guardrex** [comment]
@danroth27 Updates made.

**Pull request merged**